config/routes.rb

```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show], controller: 'user_posts'
```

Comment:
- [ ] By specifying `controller: 'user_posts'`, you are pointing Rails to `UserPostsController` which doesn't exist thereby causing the error shown below 👇. Please take it off and Rails will handle the routing correctly 👍 

![image](https://github.com/zuhebahmed88091/Blog_App_Rails/assets/46839548/008f6f7b-6258-418e-8b1b-ba7feeb97ab4)

Reply:
Okay.. done that